### ISSUE TITLE
[storage 2016-01-01] StorageAccountUpdateParameters requires sku.name when it should not

### ISSUE THREAD

**Issue by derekbekoe** [Azure/azure-rest-api-specs#370]
**TL;DR**
Sku.name is required for StorageAccounts_Update operation meaning that if you want to update a storage account, you must set Sku.name each time. The REST API does not have this limitation as you can execute the operation with a PATCH request and only modify a tag without specifying a Sku.name.

Both 2016-01-01 and 2015-06-15 of the REST APIs support not specifying Sku.name(accountType) but this is only reflected correctly in the API Spec for 2015-06-15.

---

The description for the [StorageAccounts_Update](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json) reads "The update operation can be used to update the account type, encryption, or tags for a storage account..."

With this description, I would expect to be able to run the operation with account type, encryption or tags being optional.
i.e. Account type is **optional**. This is how [StorageAccounts_Update in 2015-06-15](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2015-06-15/swagger/storage.json#L208) is defined.

In 2016-01-01 though, the Sku is set as a property of [StorageAccountUpdateParameters](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json) and in [Sku](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json), name is marked as **required**.

It looks like [StorageAccountUpdateParameters](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json) should not have Sku as a property as Sku.name is marked as required.

**Comment by lmazuel** [comment]
As I said in the Python issue, "Sku" is optional in "StorageAccountUpdateParameters", but "name" is mandatory in "Sku".
IOW, you can't provide an empty "Sku", but you can decide to not provide "Sku" at all.

**Comment by derekbekoe** [comment]
Thanks @lmazuel this clears this up.
